test/int/featureBasedSuits/pauseOverlayIsShown.test.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { testUsing } from '../fixtures/testUsing';
+import { TestProjectSpec } from '../framework/frameworkTestSupport';
+import { LaunchProject } from '../fixtures/launchProject';
+import { launchArgs } from '../testSetup';
+import { expect } from 'chai';
+import * as _ from 'lodash';
+import { utils } from 'vscode-chrome-debug-core';
+
+// There is no way to validate whether we are showing the paused overlay with puppeteer, so we look into the debug-adapter
+// log and see if we sent the proper Overlay.setPausedInDebuggerMessage message
+async function latestPausedOverlay(): Promise<string | undefined> {
+    // Wait a little to give the log file time to get written...
+    await utils.promiseTimeout(undefined, 500);
+
+    const logFilePath = launchArgs().logFilePath!;
+    const logFileContents = await utils.readFileP(logFilePath);
+    const lines = logFileContents.split('\n');
+    const lastEvent = _.findLast(lines, line => line.indexOf('Overlay.setPausedInDebuggerMessage') >= 0);
+    expect(lastEvent).to.not.equal(undefined);
+
+    // We are trying to match this string: Overlay.setPausedInDebuggerMessage\",\"params\":{ <contents here> }
+    const matches = lastEvent!.match(/Overlay\.setPausedInDebuggerMessage\\",\\"params\\":\{([^}]*)\}/);
```

Comment:
This doesn't actually test that we sent the correct message to the target, only that we logged something. 

If we truly want to test that we're sending the correct message to the target, we either need to intercept the messages in a "sniffer" or a proxy over the chrome api classes, or use a "fake target" which can validate that we sent the right message.

Reply:
The logger is basically sniffing the messages, and logging them to a file... It's true that we can have a bug on the logger, and log something we didn't send. We could also have a bug in our sniffer or proxy, and report that we sent something we didn't sent...

Reply:
> The logger is basically sniffing the messages, and logging them to a file

Fair point, but my point isn't about having bugs that cause the test to fail, but rather avoiding having our tests coupled to our logging implementation. It makes the test brittle, because there are a whole host of scenarios where a small change to the logging implementation could break the test.

You've mentioned before how important it is to make sure our tests are deterministic, and this one isn't. What happens if the test instances are extremely slow, and our 500ms timeout isn't enough?

Reply:
If that happens, we can change it to poll several times for up to 10 seconds or so...